### DISCUSSION TITLE
oh-my-posh completion powershell | Invoke-Expression throws an error

### DISCUSSION THREAD

**Discussion by Drllap** [JanDeDobbeleer/oh-my-posh#3453] [Troubleshoot] [answered]
Noticed this `completion` command in the help many and decided to try it out.

Added the command in the title to my profile but it throws this error:
![image](https://user-images.githubusercontent.com/3659986/216791117-57738e02-9998-4bc5-8013-749a1bd8aad3.png)

If I write the output of `oh-my-posh completion powershell` to a file and source it in my profile everything works just fine.

Anyone have any idea why?

**Comment by lewis-yeung** [comment, marked as answer]
@Drllap You should use `oh-my-posh completion powershell | Out-String | Invoke-Expression` which is described in the output of `oh-my-posh completion powershell -h`. :)
